src/encrypt.rs

```diff
@@ -0,0 +1,301 @@
+//! COSE Encryption
+
+use serde::{Deserialize, Serialize, Serializer, ser::SerializeSeq};
+use serde_bytes::ByteBuf;
+use serde_cbor::Error as CborError;
+use serde_cbor::Value as CborValue;
+use openssl::symm::{encrypt_aead, decrypt_aead, Cipher};
+use openssl::rand::rand_bytes;
+
+use crate::error::COSEError;
+use crate::header_map::{HeaderMap, map_to_empty_or_serialized};
+
+///
+pub enum COSEAlgorithm {
+    /// AES-GCM mode w/ 128-bit key, 128-bit tag
+    A128GCM,
+    /// AES-GCM mode w/ 192-bit key, 128-bit tag
+    A192GCM,
+    /// AES-GCM mode w/ 256-bit key, 128-bit tag
+    A256GCM,
+    /// AES-CCM mode 128-bit key, 64-bit tag, 13-byte nonce
+    AESCCM1664128,
```

Comment:
This constant is unreadable. Maybe change it to `Aes_Ccm_16_64_128` to also match the capitalizing wishes of clippy?